buildSrc/src/main/java/com/google/firebase/gradle/plugins/PublishingPlugin.kt

```diff
@@ -220,18 +220,40 @@ abstract class PublishingPlugin : Plugin<Project> {
 
     val projectNames = projectsToPublish.split(",")
 
-    val libraryGroupsToRelease =
-      allFirebaseLibraries
-        .filter { it.artifactId.get() in projectNames }
-        .map { it.libraryGroupName }
     val librariesToRelease =
-      libraryGroups
-        .filterKeys { it in libraryGroupsToRelease }
-        .flatMap { it.value }
-        .distinctBy { it.artifactId.get() }
+      getDeepTransitiveReleases(allFirebaseLibraries, projectNames, libraryGroups).toList()
+
     return ReleaseMetadata(librariesToRelease, releaseName)
   }
 
+  /**
+   * Finds all transitive libraries that need to be released so that the input list of project names
+   * can be released. This includes all project level dependencies, and anything in the same library
+   * group.
+   */
+  private fun getDeepTransitiveReleases(
```

Comment:
Could separate the loop into a `tailrec` function, and if there was a `LibraryGroup` class or `Object` (singleton) then this would be much cleaner as well:

```kotlin
private tailrec fun getDeepTransitiveReleases(
  librariesToRelease: List<FirebaseLibraryExtension>,
): List<FirebaseLibraryExtension> {
  val groupsToRelease = librariesToRelease.flatMap { it.projectLevelDeps }.map { it.libraryGroupName }
  val projects = LibraryGroups.projectsForGroups(groupsToRelease)

  return if(librariesToRelease == projects) librariesToRelease else getDeepTransitiveReleases(projects)
}
```
 
The alternative would be doing something like mapping library groups to adjacent library groups, eg;

```kotlin
LibraryGroup {
  fun getSiblingGroups(group: String): List<String>
}
// or...
class LibraryGroup {
  val siblings: List<LibraryGroup>
}
```

And then taking advantage of that by doing something like:
```kotlin
private tailrec fun getDeepTransitiveReleases(
  librariesToRelease: List<FirebaseLibraryExtension>,
): List<FirebaseLibraryExtension> {
  val groupsToRelease = librariesToRelease.flatMap { it.projectLevelDeps }.map { it.libraryGroup.siblings }
  
  return LibraryGroups.projectsForGroups(groupsToRelease)
}
```